galley/pkg/source/kube/dynamic/converter/converter.go

```diff
@@ -217,12 +217,27 @@ func kubeServiceResource(cfg *Config, _ resource.Info, name resource.FullName, _
 			Protocol: string(kube.ConvertProtocol(kubePort.Name, kubePort.Protocol)),
 		})
 	}
+
+	// TODO: Should we remove kubeServiceResource() in order to keep the original k8s Service
+	// resource (with Selector information) and compose the ServiceEntry in buildSnapshot()?
+	// Store the service selector in annotation which will be later used to update the authentication
+	// policy to use label selectors instead of service names.
```

Comment:
Why do you need annotation? Can we just check if "selector" is there, use "selector", otherwise, use "service"?

Reply:
The reason I need to modify here is that `kubeServiceResource()` converts the k8s `Service` resource into the Istio `ServiceEntry`, the generated `ServiceEntry` doesn't have the original selector information anymore. 

Later in `buildSnapshot()`, we only have access to the `ServiceEntry` which doesn't have the original selector.

The `kubeServiceResource()` converts the k8s `Service` to Istio `ServiceEntry` but still storing it using the k8s service key `k8s/core/v1/services`. This seems a little bit hack to me and that's why I'm asking should we actually refactor this in order to keep the original k8s `Service` and do the conversion later in `buildSnapshot()`.